src/main/kotlin/nmcp/internal/utils.kt

```diff
@@ -107,3 +109,23 @@ internal fun Project.registerPublishToCentralPortalTasks(
     }
 }
 
+fun Project.createCentralPortalOptions(action: Action<CentralPortalOptions>): CentralPortalOptions {
+    val centralPortalOptions = objects.newInstance(CentralPortalOptions::class.java)
+    action.execute(centralPortalOptions)
+
+    project.gradle.taskGraph.whenReady {
+        if (it.allTasks.any {
+                it is NmcpPublishWithPublisherApiTask
```

Comment:
`hasTask(Task)` might be more efficient than scanning all the tasks

Reply:
I think [both of them are equally bad](https://github.com/gradle/gradle/issues/33344) 😅

Reply:
Are you sure `(Task)` overload is equally bad?

Reply:
Nope, I didn't check this, will need to deep dive into this.

Reply:
`hasTask(Task)` sadly doesn't have a lazy version. Is it fine calling `taskProvider.get()` unconditionnally for this?

Reply:
For the record, I switched to `hasTask(String)` [here](https://github.com/GradleUp/nmcp/pull/164) because `allTasks` is not isolated-projects compatible.